src/command.directive.ts

```diff
@@ -8,37 +8,77 @@ import {
 	ElementRef,
 	Inject,
 	Renderer2,
-	ChangeDetectorRef
+	ChangeDetectorRef,
+	ViewContainerRef,
 } from "@angular/core";
 import { Subscription, Observable, EMPTY } from "rxjs";
 import { tap, merge } from "rxjs/operators";
 
 import { CommandOptions, COMMAND_CONFIG, COMMAND_DEFAULT_CONFIG } from "./config";
-import { ICommand } from "./command";
+import { Command } from "./command";
+import { isCommand, isCommandCreator } from "./command.util";
+import { CommandCreator, ICommand } from "./command.model";
 
 /**
+ * Controls the state of a component in sync with `Command`.
  *
- * ### Example with options
+ * ### Most common usage
+ * ```html
+ * <button [command]="saveCmd">Save</button>
+ * ```
+ *
+ *
+ * ### Usage with options
  * ```html
  * <button [command]="saveCmd" [commandOptions]="{executingCssClass: 'in-progress'}">Save</button>
  * ```
+ *
+ *
+ * ### Usage with params
+ * This is useful for collections (loops) or using multiple actions with different args.
+ * *NOTE: This will share the `isExecuting` when used with multiple controls.*
+ *
+ * #### With single param
+ *
+ * ```html
+ * <button [command]="saveCmd" [commandParams]="{id: 1}">Save</button>
+ * ```
+ * *NOTE: if you have only 1 argument, and it should be an array, it should be as enclosed within an array e.g. `[['apple', 'banana']]`,
+ * else it will spread and you will `arg1: "apple", arg2: "banana"`*
+ *
+  * #### With multi params
+ * ```html
+ * <button [command]="saveCmd" [commandParams]="[{id: 1}, 'hello', hero]">Save</button>
+ * ```
+ *
+ * ### Usage with Command Creator
+ * This is useful for collections (loops) or using multiple actions with different args, whilst not sharing `isExecuting`.
+ *
+ *
+ * ```html
+ * <button [command]="{execute: removeHero$, canExecute: isValid$, params: [hero, 1337, 'xx']}">Save</button>
+ * ```
+ *
  */
 @Directive({
 	selector: "[command]",
 })
 export class CommandDirective implements OnInit, OnDestroy {
-	@Input() command!: ICommand;
+	@Input("command") commandInput!: ICommand | CommandCreator | undefined;
 	@Input() commandOptions!: CommandOptions;
+	@Input() commandParams: any;
```

Comment:
```suggestion
	@Input() commandParams: any[];
```

Reply:
I m not sure if this need to be an array:
since even if its one item it needs to be an array

Reply:
no, both of them are supported, i will improve doc, thanks. Will change it to `any | any[]`